src/log.js

```diff
@@ -22,6 +22,30 @@ import {getModeObject} from './mode-object';
 const start = new Date().getTime();
 
 /**
+ * @const {!Array<!Error>}
+ */
+const asyncErrorsThrownForTesting = [];
+
+
+/**
+ * @param {!Error} error
+ */
+function recordAsyncErrorForTesting(error) {
```

Comment:
> provide API to force throwing async thrown errors in tests

That'll break the assumption of our source code. Just having a notification system we can check though would work.